openshift/image-inspector-template.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: "v1"
+kind: "Template"
+metadata:
+  name: "image-inspector-scan"
+  annotations:
+    description:
+      Template for scanning a container image. Requires privileged permissions.
+    tags: infrastructure
+objects:
+- apiVersion: "v1"
+  kind: "Pod"
+  metadata:
+    name: "${NAME}"
+    labels:
+      name: "${NAME}"
+    annotations:
+      openshift.io/scc: "privileged"
+      openshift.io/image-scan-target: "${TARGET_IMAGE}"
+  spec:
+    containers:
+    - name: "image-inspector"
+      image: "${IMAGE_REGISTRY}/${IMAGE_NAME}:${IMAGE_TAG}"
+      imagePullPolicy: "Always"
+      command:
+      - "/usr/bin/image-inspector"
+      - "--chroot"
+      - "--username=${SERVICE_ACCOUNT}"
+      - "--password-file=${TOKEN_FILE}"
+      - "--image=${TARGET_IMAGE}"
+      - "--scan-type=openscap"
+      - "--cve-url=${CVE_URL}"
+      - "--openscap-html-report"
+      - "--serve=:8080"
+      ports:
+      - containerPort: 8080
+        protocol: "TCP"
+      volumeMounts:
+      - name: "docker-socket"
+        mountPath: "${DOCKER_SOCKET}"
+      securityContext:
+        privileged: true
+      readinessProbe:
+        httpGet:
+          path: "/healthz"
+          port: 8080
+        initialDelaySeconds: 15
+        periodSeconds: 5
+    nodeSelector: "${{NODE_SELECTOR}}"
+    automountServiceAccountToken: true
+    serviceAccountName: "${SERVICE_ACCOUNT}"
+    restartPolicy: "Never"
+    volumes:
```

Comment:
I wonder if we should define an `emptyDir` volume here for where we extract the image.

Reply:
what would be the use case for that?

Reply:
@ilackarms to not abuse the COW docker storage for the image extraction

Reply:
FYI, empty dir is quota protected in some environments.  We need to keep that in mind.  @abhgupta can provide details

Reply:
Anyone (@pweil- @mfojtik @joelsmith @legionus), any suggestion on where we can properly extract the image?
I'd rather not have a dependency on a PV for a temporary "cache". OTOH it seems hard to make assumptions on `emptyDir`.

Reply:
I think you'd need to use a PV as a scratch space (whether that be `emptyDir` or something else).  The prereq is that the admin knows it has enough space.  There is also the option of [local storage](https://github.com/kubernetes-incubator/external-storage/tree/master/local-volume) that is alpha in 1.7/1.8 kube that we should keep in mind to see if it fits the requirements.